strawberryfields/apps/sample.py

```diff
@@ -0,0 +1,240 @@
+# Copyright 2019 Xanadu Quantum Technologies Inc.
+r"""
+Sampling functions
+==================
+
+**Module name:** :mod:`strawberryfields.future.apps.sample`
+
+.. currentmodule:: strawberryfields.future.apps.sample
+
+This module provides functionality for generating samples from both a quantum device and the
+uniform distribution.
```

Comment:
In this case, the sampling is with respect to graph structures, so I'm not sure I follow the structure of having this outside the graph package?

Reply:
:thinking: I'd say for `quantum_sampler` the sampling is with respect to any _symmetric_ matrix. Whether this is sampling from a graph could be debated, e.g. it could be a graph with complex weights. Things like point processes, which encode a grid of distances, would access this module, but I wouldn't see a direct link to a graph. I'd prefer to keep "graph" sampling to the ones which have real and symmetric adjacency matrices - which is what's imposed in `to_networkx_graph` or now `validate_graph`.

On the other hand, this module outputs samples that are of `len(modes)` and are zeros/ones/ints. Hence, the samples output aren't graphs (although simple to convert). Instead, in `graph.sample`, the output samples are subgraphs, i.e. a list of ints selecting nodes of a subgraph.

Reply:
Ah, I think I see. I'm worked a lot with complex weighted undirected graphs in my PhD, so to me, almost any square matrix is a graph!